src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -4405,10 +4353,17 @@ def _contains_intrinsic_reduction(pnodes):
             # variable in the logical-array expression must be an array for
```

Comment:
You may be able to simplify this: just identify the child of `fake_parent` that represents "logical_expr" and ask for its `datatype`.

Reply:
I'm not sure - I need to replace any of the non-array syntax'd arrays in this logical_expr (since there can be multiples I think?) so just finding the datatype isn't sufficient, or am I misunderstanding.

Reply:
Ah yes. I'd forgotten that you need the list of `arrays` in order to then update them. As with my other suggestion, it may be possible to apply `Reference2ArrayRangeTrans` to each of `fake_parent.walk(Reference)` and then you won't need your new bit of code (and the complexity is kept within that transformation).